seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/functions/ArrayFunction.java

```diff
@@ -101,36 +95,31 @@ public static Object[] array(List<Object> args) {
     }
 
     public static ArrayType castArrayTypeMapping(Function function, SeaTunnelRowType inputRowType) {
-        return castArrayTypeMapping(getFunctionArgs(function, inputRowType));
-    }
+        List<Expression> expressions = CommonFunction.getExpressions(function);
 
-    public static ArrayType castArrayTypeMapping(List<Class<?>> args) {
-        if (args == null || args.isEmpty()) {
+        if (expressions.isEmpty()) {
             return ArrayType.STRING_ARRAY_TYPE;
         }
 
-        Class<?> arrayType = getClassType(args);
-        return getSeaTunnelDataType(arrayType);
+        SeaTunnelDataType<?> elementType = null;
+        for (Expression expr : expressions) {
+            SeaTunnelDataType<?> t = CommonFunction.toType(expr, inputRowType);
+            elementType = CommonFunction.unifyCollectionType(elementType, t);
+        }
+        if (elementType == null) {
+            elementType = BasicType.STRING_TYPE;
+        }
+        return createArrayType(elementType);
     }
 
-    private static ArrayType getSeaTunnelDataType(Class<?> clazz) {
-        String className = clazz.getSimpleName();
-        switch (className) {
-            case "Integer":
-                return ArrayType.INT_ARRAY_TYPE;
-            case "Double":
-                return ArrayType.DOUBLE_ARRAY_TYPE;
-            case "Boolean":
-                return ArrayType.BOOLEAN_ARRAY_TYPE;
-            case "Long":
-                return ArrayType.LONG_ARRAY_TYPE;
-            case "float":
-                return ArrayType.FLOAT_ARRAY_TYPE;
-            case "short":
-                return ArrayType.SHORT_ARRAY_TYPE;
-            default:
-                return ArrayType.STRING_ARRAY_TYPE;
-        }
+    static ArrayType createArrayType(SeaTunnelDataType<?> elementType) {
+        if (elementType == BasicType.INT_TYPE) return ArrayType.INT_ARRAY_TYPE;
+        if (elementType == BasicType.LONG_TYPE) return ArrayType.LONG_ARRAY_TYPE;
+        if (elementType == BasicType.DOUBLE_TYPE) return ArrayType.DOUBLE_ARRAY_TYPE;
+        if (elementType == BasicType.FLOAT_TYPE) return ArrayType.FLOAT_ARRAY_TYPE;
+        if (elementType == BasicType.SHORT_TYPE) return ArrayType.SHORT_ARRAY_TYPE;
+        if (elementType == BasicType.BOOLEAN_TYPE) return ArrayType.BOOLEAN_ARRAY_TYPE;
+        return ArrayType.STRING_ARRAY_TYPE;
```

Comment:
So if `elementType` is `ArrayType.STRING_ARRAY_TYPE`. The method will return `ARRAY<STRING>`? Why not `ARRAY<ARRAY<STRING>>`? We should support create any array type. So I think you should do some upgrade for ArrayType class.